app/org/sagebionetworks/bridge/services/UploadSchemaService.java

```diff
@@ -27,6 +27,27 @@ public void setUploadSchemaDao(UploadSchemaDao uploadSchemaDao) {
     }
 
     /**
+     * Service handler for creating a new schema revision, using V4 API semantics. See
+     * {@link UploadSchemaDao#createSchemaRevisionV4}
+     *
+     * @param studyId
+     *         study that will contain the schema
+     * @param uploadSchema
+     *         schema to create
+     * @return the created schema revision
+     */
+    public UploadSchema createSchemaRevisionV4(StudyIdentifier studyId, UploadSchema uploadSchema) {
+        // validate schema
+        if (uploadSchema == null) {
+            throw new InvalidEntityException("Upload schema can't be null");
```

Comment:
I think some precondition checks are enough here. The controller at least will create an object even if no fields are set, so this would be some kind of developer programming error.